cupyx/scipy/signal/signaltools.py

```diff
@@ -0,0 +1,78 @@
+import cupy
+
+
+def _numeric_arrays(arrays, kinds='buifc'):
+    if type(arrays) == cupy.ndarray:
+        return arrays.dtype.kind in kinds
+    for array in arrays:
+        if array.dtype.kind not in kinds:
+            return False
+    return True
+
+
+def choose_conv_method(in1, in2, mode='full'):
+    """Find the fastest convolution/correlation method.
+
+    Args:
+        in1 (cupy.ndarray): first input.
+        in2 (cupy.ndarray): second input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        str: A string indicating which convolution method is fastest,
+         either ‘direct’ or ‘fft’.
+
+    .. warning::
+        This function currently doesn't support measure option,
+        nor multidimensional inputs.
+
+    .. seealso:: :func:`scipy.signal.choose_conv_method`
+
+    """
+    if in1.ndim != 1 or in2.ndim != 1:
+        raise NotImplementedError('Only 1d inputs are supported currently')
+
+    if any([_numeric_arrays([x], kinds='ui') for x in [in1, in2]]):
+        max_value = int(cupy.abs(in1).max()) * int(cupy.abs(in2).max())
+        max_value *= int(min(in1.size, in2.size))
+        if max_value > 2 ** cupy.finfo('float').nmant - 1:
+            return 'direct'
+
+    if _numeric_arrays([in1, in2], kinds='b'):
+        return 'direct'
+
+    if _numeric_arrays([in1, in2]):
```

Comment:
`_numeric_arrays([in1, in2])` always returns `True` because `cupy.ndarray` supports only for numeric elements.